src/common.hpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2022 YADRO.
+ */
+
+#pragma once
+
+#include <sdbusplus/bus.hpp>
+#include <sdbusplus/bus/match.hpp>
+
+#include <functional>
+#include <map>
+#include <string>
+
+extern "C"
+{
+#include <i2c/smbus.h>
+}
+
+/**
+ * @class PowerState
+ *
+ * This class provides an interface to check the current power state,
+ * and to register a function that gets called when there is a power
+ * state change.  A callback can be passed in using the constructor,
+ * or can be added later using addCallback().
+ *
+ * @note the code is based on power state monitor from phosphor-fan-presence
+ */
+class PowerState
```

Comment:
It seems to me these should be split  onto separate headers/sources.
This would allow to use `PowerState` without linking with i2c in other stuff of this project.
For example, in https://github.com/YADRO-KNS/obmc-yadro-hardware/pull/15